src/scitacean/client.py

```diff
@@ -249,17 +249,16 @@ def upload_new_dataset_now(self, dataset: Dataset) -> Dataset:
         dataset = dataset.replace(
             source_folder=self._expect_file_transfer().source_folder_for(dataset)
         )
-        dataset.validate()
+        dset_model = dataset.make_upload_model()
+        self.scicat.validate_dataset_model(dset_model)
         # TODO skip if there are no files
         with self._connect_for_file_upload(dataset) as con:
             # TODO check if any remote file is out of date.
             #  if so, raise an error. We never overwrite remote files!
             uploaded_files = con.upload_files(*dataset.files)
             dataset = dataset.replace_files(*uploaded_files)
             try:
-                finalized_model = self.scicat.create_dataset_model(
-                    dataset.make_upload_model()
-                )
+                finalized_model = self.scicat.create_dataset_model(dset_model)
```

Comment:
This is technically wrong. `dataset` got changed above to include the uploaded file info. So you need to generate a new model here. The only difference this makes is that the size may change. Because differences in the filesystem of remote and local can lead to different file sizes. And the old code used the remote size, but the new code uses the local size.

This doesn't get caught by the tests because they only run on one filesystem.

Reply:
Ok. Do we still want to validate before `self._connect_for_file_upload(dataset)` as we are now, or can we defer validation to here so that we only make the model once?

I am guessing making the model is cheap enough that this is probably not an issue?

Reply:
It's cheap compared to http requests. So I wouldn't worry.

You need to make the model for upload after the file upload. So it would then defer validation until then which is the opposite of what we want.